Terms cleanup

vlukes opened the pull request:
Changelist:
- 'd_diffusion_sa' renamed to 'd_sd_diffusion', term moved to terms_diffusion.py
-  removed files: terms_acoustic.*
- 'dw_lin_elastic_iso' is now the shortcut to 'dw_lin_elastic' + stiffness_from_lame(), the C code related to 'dw_lin_elastic_iso' removed
- examples: 'dw_lin_elastic_iso' replaced by 'dw_lin_elastic'
- tests: the same as above, _iso term is tested in test_elasticity_small_strain.py
- 'di_surface_moment' renamed to 'd_surface_moment',  there are no important differences between 'di_' a 'd_',  there is the only one 'di_' term
- term table divided into: basic, sensitivity, large deformation, special terms

94 test file(s) executed in 143.03 s, 0 failure(s) of 156 test(s)

rc commented on doc/terms_overview.rst:
```diff
@@ -105,17 +105,7 @@ Term names are (usually) prefixed according to the following conventions:
-     - `'qp'`
-     - terms having all arguments known, the result are the values in
-       quadrature points of elements
+     - terms having all arguments known, the result is value of the integral
```
the result is the value of the term integral evaluation.

rc commented on script/gen_term_table.py:
```diff
@@ -50,7 +50,7 @@ def create_parser(slist, current_section):
 header = """
 .. tabularcolumns:: |p{0.2\linewidth}|p{0.2\linewidth}|p{0.6\linewidth}|
-.. list-table:: Table of all terms.
+.. list-table:: XXX terms
```
Why not use ``%s`` here and ``%`` below instead of ``header.replace()``?

rc commented:
Thank you, very nice clean up!

A few notes:

- I am not sure about replacing ``dw_lin_elastic_iso`` everywhere - maybe it could be left in (some of) the examples, where the 'eval' mode of ``dw_lin_elastic`` is not needed?
- As a matter of fact, ``LinearElasticIsotropicTerm`` should inherit from ``LinearElasticTerm``, and in its ``.get_fargs()`` only transform the arguments and call ``LinearElasticTerm.get_fargs()``. The same for ``.get_eval_shape()``. Note that I somehow forgot to remove ``LinearElasticTerm.check_shapes()`` in d1e165ef9b8b5f8e6bbad91e0114263c7fb8d8ec. You can remove it in the process :)
- Some new additions in ``tests/test_elasticity_small_strain.py`` have extra white space in arguments lists.
- I made some comments to lines of the term table commit. Also, there seems to be a problem with choosing where each term should go - some terms are in several tables (e.g. ``d_sd_st_pspg_p`` and other sensitivity terms), some in none (``d_sd_diffusion``). There is a typo in the commit message.
- Now also d_* terms seem like good candidates to be renamed to ev_* :) (later?)

vlukes commented:
On 2016-03-29 15:50, Robert Cimrman wrote:
> Thank you, very nice clean up!
>
> A few notes:
>
>   * I am not sure about replacing |dw_lin_elastic_iso| everywhere -
>     maybe it could be left in (some of) the examples, where the 'eval'
>     mode of |dw_lin_elastic| is not needed?
My intetion was to prepare 'dw_lin_elastic_iso' term for removing, now 
it can be done easily.

>   * As a matter of fact, |LinearElasticIsotropicTerm| should inherit
>     from |LinearElasticTerm|, and in its |.get_fargs()| only transform
>     the arguments and call |LinearElasticTerm.get_fargs()|. The same for
>     |.get_eval_shape()|. Note that I somehow forgot to remove
>     |LinearElasticTerm.check_shapes()| in d1e165e
>     <https://github.com/sfepy/sfepy/commit/d1e165ef9b8b5f8e6bbad91e0114263c7fb8d8ec>.
>     You can remove it in the process :)
OK, check_shapes() removed, LinearElasticIsotropicTerm inherits from 
LinearElasticTerm

>   * Some new additions in |tests/test_elasticity_small_strain.py| have
>     extra white space in arguments lists.
fixed

>   * I made some comments to lines of the term table commit. Also, there
>     seems to be a problem with choosing where each term should go - some
>     terms are in several tables (e.g. |d_sd_st_pspg_p| and other
>     sensitivity terms), some in none (|d_sd_diffusion|). There is a typo
>     in the commit message.
fixed

>   * Now also d_* terms seem like good candidates to be renamed to ev_*
>     :) (later?)
I think, it could be done later :-)

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/sfepy/sfepy/pull/315#issuecomment-202902240>Web Bug
> from
> https://github.com/notifications/beacon/ABT7_ZkKH2UklEnOovM9WmImj6QHXSg0ks5pyS5BgaJpZM4H6sGW.gif
>

rc commented:
On 03/30/2016 06:33 PM, Vladimir Lukes wrote:
> On 2016-03-29 15:50, Robert Cimrman wrote:
>> Thank you, very nice clean up!
>>
>> A few notes:
>>
>>    * I am not sure about replacing |dw_lin_elastic_iso| everywhere -
>>      maybe it could be left in (some of) the examples, where the 'eval'
>>      mode of |dw_lin_elastic| is not needed?
> My intetion was to prepare 'dw_lin_elastic_iso' term for removing, now
> it can be done easily.

OK.

One more detail: could you explicitly name the arguments of 
``stiffness_from_lame()`` in the changed examples and other files (e.g. 
``stiffness_from_lame(3, lam=1e1, mu=1e0)``), so that it is clear which value 
is which without looking it up in the docs?

>>    * As a matter of fact, |LinearElasticIsotropicTerm| should inherit
>>      from |LinearElasticTerm|, and in its |.get_fargs()| only transform
>>      the arguments and call |LinearElasticTerm.get_fargs()|. The same for
>>      |.get_eval_shape()|. Note that I somehow forgot to remove
>>      |LinearElasticTerm.check_shapes()| in d1e165e
>>      <https://github.com/sfepy/sfepy/commit/d1e165ef9b8b5f8e6bbad91e0114263c7fb8d8ec>.
>>      You can remove it in the process :)
> OK, check_shapes() removed, LinearElasticIsotropicTerm inherits from
> LinearElasticTerm

Thanks! Instead of creating the mapping, you could use ``self.region.dim`` for 
the space dimension.

>>    * Some new additions in |tests/test_elasticity_small_strain.py| have
>>      extra white space in arguments lists.
> fixed
>
>>    * I made some comments to lines of the term table commit. Also, there
>>      seems to be a problem with choosing where each term should go - some
>>      terms are in several tables (e.g. |d_sd_st_pspg_p| and other
>>      sensitivity terms), some in none (|d_sd_diffusion|). There is a typo
>>      in the commit message.
> fixed
>
>>    * Now also d_* terms seem like good candidates to be renamed to ev_*
>>      :) (later?)
> I think, it could be done later :-)

Sure :)

Except my above comments, all looks fine!

r.

vlukes commented:
On 2016-03-31 09:06, Robert Cimrman wrote:
> One more detail: could you explicitly name the arguments of
> ``stiffness_from_lame()`` in the changed examples and other files (e.g.
> ``stiffness_from_lame(3, lam=1e1, mu=1e0)``), so that it is clear which
> value
> is which without looking it up in the docs?

ok, done.


>  >> * As a matter of fact, |LinearElasticIsotropicTerm| should inherit
>  >> from |LinearElasticTerm|, and in its |.get_fargs()| only transform
>  >> the arguments and call |LinearElasticTerm.get_fargs()|. The same for
>  >> |.get_eval_shape()|. Note that I somehow forgot to remove
>  >> |LinearElasticTerm.check_shapes()| in d1e165e
>  >>
> <https://github.com/sfepy/sfepy/commit/d1e165ef9b8b5f8e6bbad91e0114263c7fb8d8ec>.
>  >> You can remove it in the process :)
>  > OK, check_shapes() removed, LinearElasticIsotropicTerm inherits from
>  > LinearElasticTerm
>
> Thanks! Instead of creating the mapping, you could use
> ``self.region.dim`` for
> the space dimension.

fixed

rc commented:
Thanks!
